src/ZEO/ClientStorage.py

```diff
@@ -799,40 +796,35 @@ def tpc_begin(self, txn, tid=None, status=' '):
             raise POSException.ReadOnlyError()
 
         try:
-            tbuf = txn.data(self)
+            tbuf = txn.__buf
         except AttributeError:
-            # Gaaaa. This is a recovery transaction. Work around this
```

Comment:
Except that the reason transactions need that is because there may be more than one data manager, but for transaction metadata and recovery transaction data, TCBOO client.